src/custom/ecospheres/components/datasets/DatasetAddToBouquetModal.vue

```diff
@@ -0,0 +1,141 @@
+<script setup lang="ts">
+import type { DatasetV2 } from '@etalab/data.gouv.fr-components'
+import { ref, onMounted, computed } from 'vue'
+import { toast } from 'vue3-toastify'
+import { useLoading } from 'vue-loading-overlay'
+
+import Tooltip from '@/components/TooltipWrapper.vue'
+import { Availability, type DatasetProperties, type Topic } from '@/model'
+import { useTopicStore } from '@/store/TopicStore'
+
+import DatasetPropertiesTextFields from '../forms/dataset/DatasetPropertiesTextFields.vue'
+
+const props = defineProps({
+  show: {
+    type: Boolean,
+    default: false
+  },
+  dataset: {
+    type: Object as () => DatasetV2,
+    required: true
+  }
+})
+
+const emit = defineEmits(['update:show'])
+const loader = useLoading()
+const topicStore = useTopicStore()
+
+const bouquets = topicStore.userTopics
+const datasetProperties = ref<DatasetProperties>({
+  title: '',
+  purpose: '',
+  id: props.dataset.id,
+  uri: `/datasets/${props.dataset.id}`,
+  availability: Availability.LOCAL_AVAILABLE
+})
+const selectedBouquetId = ref(null)
+
+const bouquetOptions = computed(() => {
+  // sort on a copy to avoid modifying bouquets.value in a computed property
+  const sortedBouquets = [...bouquets.value].sort((a, b) => {
+    return (
+      new Date(b.last_modified).getTime() - new Date(a.last_modified).getTime()
+    )
+  })
+  return sortedBouquets.map((bouquet) => {
+    return {
+      value: bouquet.id,
+      text: bouquet.name,
+      disabled: isDatasetInBouquet(bouquet)
+    }
+  })
+})
+
+const isValid = computed(() => {
+  return (
+    datasetProperties.value.title.trim() !== '' &&
+    datasetProperties.value.purpose.trim() !== '' &&
+    !!selectedBouquetId.value
+  )
+})
+
+const modalActions = computed(() => {
+  return [
+    {
+      label: 'Valider',
+      disabled: !isValid.value,
+      onClick: () => submit()
+    },
+    {
+      label: 'Annuler',
+      secondary: true,
+      onClick: () => closeModal()
+    }
+  ]
+})
+
+const isDatasetInBouquet = (bouquet: Topic): boolean => {
+  const datasetsProperties =
+    bouquet.extras['ecospheres:datasets_properties'] || []
+  return datasetsProperties.some(
+    (datasetProps) => datasetProps.id === props.dataset.id
+  )
+}
+
+const submit = async () => {
+  if (selectedBouquetId.value === null) {
+    throw Error('Trying to attach to bouquet without id')
+  }
+  const bouquet = topicStore.get(selectedBouquetId.value)
+  if (bouquet === undefined) {
+    throw Error('Bouquet not in store')
+  }
+  const newDatasetsProperties =
+    bouquet.extras['ecospheres:datasets_properties'] || []
+  newDatasetsProperties.push(datasetProperties.value)
+  bouquet.extras['ecospheres:datasets_properties'] = newDatasetsProperties
+  await topicStore.update(bouquet.id, {
+    id: bouquet.id,
+    tags: bouquet.tags,
+    extras: bouquet.extras
+  })
+  toast('Jeu de données ajouté avec succès', { type: 'success' })
```

Comment:
```suggestion
  toast(`Jeu de données ajouté avec succès au bouquet "${bouquet.name}"`, { type: 'success' })
```
Peut-être faut-il racourcir `bouquet.name` si trop long ?